### PR TITLE
Update the Block List documentation to show how to add an angularJS controller

### PR DESCRIPTION
**Pull request by marcemarc**
to the backoffice view

with example to display an image from a media picker

### CODE REVIEW

File: Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md [hunk 1]
```diff
@@ -209,4 +209,86 @@ The following example displays the property with the alias `headline` together w
 
 If you'd like to display properties of `settings`, you can access these by `block.settingsData.myPropertyAlias`.
 
-See this forum post to [display an image or add other custom behaviour to your custom backoffice view](https://our.umbraco.com/forum/using-umbraco-and-getting-started/103560-image-in-custom-block-editor-view#comment-323696).
+### Adding custom implementation to your View
+
+To achieve this you need to add a custom angularJS controller to your custom view, using the ng-controller attribute:
+```
```
Comment (sofietoft) [inline Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md:215]:
```suggestion
```html
```

File: Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md [hunk 2]
```diff
@@ -209,4 +209,86 @@ The following example displays the property with the alias `headline` together w
...
+<div ng-controller="customBlockController" ng-click="block.edit()"> 
+    <h2 ng-bind="block.data.headline"></h2>
+    <p ng-bind="block.data.description"></p>
+</div>
+```
+
+Create a folder inside the App_Plugins folder called 'CustomBlockView' or something more meaningful for your implementation.
+
+Inside this folder create two files, one called package.manifest that contains the following:
```
Comment (sofietoft) [inline Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md:224]:
```suggestion
Inside this folder create two files,. The first file should be called package.manifest that contain the following:
```

File: Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md [hunk 3]
```diff
@@ -209,4 +209,86 @@ The following example displays the property with the alias `headline` together w
...
+```
```
Comment (sofietoft) [inline Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md:225]:
```suggestion
```json
```

File: Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md [hunk 4]
```diff
@@ -209,4 +209,86 @@ The following example displays the property with the alias `headline` together w
...
+{
+  "javascript": [
+    "~/App_Plugins/CustomBlockView/customBlock.controller.js"    
+  ]  
+}
+```
+Umbraco will parse all package.manifest files and load any resources they reference into the backoffice during startup.
+
+and secondly add a javascript file called
```
Comment (sofietoft) [inline Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md:234]:
```suggestion
The second file should be a javascript file called
```

File: Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md [hunk 5]
```diff
@@ -209,4 +209,86 @@ The following example displays the property with the alias `headline` together w
...
+
+**customBlock.controller.js**
+
+(to match the file referenced in your package.manifest file)
+
+this file then should register your 'customBlockController' mentioned in your view ng-controller attribute with Umbraco's angular module's controllers
```
Comment (sofietoft) [inline Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md:240]:
```suggestion
This file then should register your 'customBlockController' mentioned in your view ng-controller attribute with Umbraco's angular module's controllers
```

File: Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md [hunk 6]
```diff
@@ -209,4 +209,86 @@ The following example displays the property with the alias `headline` together w
...
+```
```
Comment (sofietoft) [inline Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md:241]:
```suggestion
```javascript
```

File: Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md [hunk 7]
```diff
@@ -209,4 +209,86 @@ The following example displays the property with the alias `headline` together w
...
+angular.module("umbraco").controller("customBlockController", function ($scope) {
+// you can do your custom functionality here!
+}
+```
+#### Example: Displaying an image from a Media Picker
+Your block may enable you to 'pick' an image for use as the background for a particular block or to display as part of the block layout, but if you try to display this image directly in the view from the property `block.data.image` you'll see the unique id and not the image.
```
Comment (sofietoft) [inline Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md:247]: Sentence was a tat too long - tried splitting it up. Hope it still makes sense 🤞

```suggestion
Your block may enable you to 'pick' an image for use as the background for a particular block or to display as part of the block layout. If you try to display this image directly in the view from the property `block.data.image` you'll see the unique id and not the image.
```

File: Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md [hunk 8]
```diff
@@ -209,4 +209,86 @@ The following example displays the property with the alias `headline` together w
...
+
+We'll need to use the Id in our custom angularJS controller to get the ImageUrl to display in our Backoffice Block Editor View.
+
+With the setup of files above, we would amend our customBlock.controller.js file, injecting the mediaResource to retrieve the image from the id:
+```
```
Comment (sofietoft) [inline Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md:252]:
```suggestion
```javascript
```

File: Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md [hunk 9]
```diff
@@ -209,4 +209,86 @@ The following example displays the property with the alias `headline` together w
...
+            .then(function (media) {
+                console.log(media);
+                //set a property on the 'scope' called imageUrl for the returned media object's mediaLink
+                $scope.imageUrl = media.MediaLink;
+    });    
+});
+```
+Update the View to use the 'imageUrl' property to display the image:
+
+```
```
Comment (sofietoft) [inline Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md:268]:
```suggestion
```html
```

File: Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md [hunk 10]
```diff
@@ -209,4 +209,86 @@ The following example displays the property with the alias `headline` together w
...
+<div ng-controller="customBlockController" ng-click="block.edit()">
+    <h2 ng-bind="block.data.headline"></h2>
+    <img src="{{imageUrl}}" />
+    <p ng-bind="block.data.description"></p>
+</div>
+```
+If you need to use a specific crop, you can inject the `imageUrlGeneratorResource` resource, which has a `getCropUrl(mediaPath, width, height, imageCropMode, animationProcessMode)` method:
+```
```
Comment (sofietoft) [inline Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md:276]:
```suggestion
```javascript
```

**Review by sofietoft — CHANGES_REQUESTED** [review]
Looks great @marcemarc !

I've made some suggestions here and there.
Mainly adding syntax highlighting to the code samples.

**Comment by marcemarc** [comment]
Thanks @sofietoft !! I think I have updated them all.

**Review by sofietoft — APPROVED** [review]
Great! Thanks @marcemarc 😄 

Let's get it merged 💪

**Pull request merged**